src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -65,13 +66,27 @@ internal bool IsDeltaOfT
             }
         }
 
+        internal bool IsChangedObjectCollection
+        {
+            get
+            {
+                if (!_isChangedObjectCollection.HasValue)
+                {
+                    _isChangedObjectCollection = ResourceType != null && TypeHelper.IsGenericType(ResourceType) && 
+                        (ResourceType.GetGenericTypeDefinition() == typeof(EdmChangedObjectCollection<>) || ResourceType.GetGenericTypeDefinition() == typeof(EdmChangedObjectCollection));
```

Comment:
don't you need to group IsGenericType and typeof(EdmChangedObjectCollection<>)? otherwise typeof(EdmChangedObjectCollection) would never be evaluated.